DOCS-780: Add bounding box labels (#1424)
viamrobotics/docs: docs/manage/data/label.md

---
title: "Label Data"
linkTitle: "Label Data"
description: "Label image data on the DATA page in the Viam Cloud."
weight: 39
type: "docs"
tags: ["data management", "cloud", "sync"]
images: [ "/manage/data/img/label-dog.gif" ]
# SME: Tahiya Salam and Alexis Wei
---

To label data in a dataset, go to the [**DATA** tab](https://app.viam.com/data/view) in the Viam app.

On the **Images** subtab, you can filter available images, using the **Filtering** menu and select the attributes that match where, how, and when the data was collected.

{{< alert title="Note" color="note" >}}
Datasets are views, not materialized.
If the underlying data matching the filter changes because data is deleted or more data is added, the dataset will also change.

Therefore the data you are viewing may change as you label and train on the dataset.
{{< /alert >}}

You can label you data with:

- [Bounding boxes](#bounding-boxes), which you can use to [train object detection models](/manage/ml/train-model/#train-a-model).
- [Image tags](#image-tags), which you can use to [image classification models](/manage/ml/train-model/#train-a-model).

### Bounding boxes

> You can create bounding boxes for objects in your images, which then allow you to [create object detection models](/manage/ml/train-model/#train-a-model).

Author: ```suggestion
You can create one or more bounding boxes for objects in each image.
If you label an entire dataset, you can use these labels to [create object detection models](/manage/ml/train-model/#train-a-model).
```

For example, if you would like to create a model that detects a dog in an image, add bounding boxes around the dog in each of your images and add or select the label `dog`.

To add a bounding box, click on the image you want to add the bounding box to.
More information about the image will open up to the right side.
Select the **Bounding Box** option.
Choose an existing label or create a new label.
Click on the image where you would like to add the bounding box and drag to where the bounding box should end.

> {{<gif webm_src="../img/label-dog.webm" mp4_src="../img/label-dog.mp4" alt="Add bounding box label around dog in an image">}}

Author: ```suggestion
{{<gif webm_src="../img/label-dog.webm" mp4_src="../img/label-dog.mp4" alt="Add a bounding box around the dog in an image">}}
```

---
title: "Label Data"
linkTitle: "Label Data"
description: "Label image data on the DATA page in the Viam Cloud."
weight: 39
type: "docs"
tags: ["data management", "cloud", "sync"]
images: [ "/manage/data/img/label-dog.gif" ]
# SME: Tahiya Salam and Alexis Wei
---

To label data in a dataset, go to the [**DATA** tab](https://app.viam.com/data/view) in the Viam app.

On the **Images** subtab, you can filter available images, using the **Filtering** menu and select the attributes that match where, how, and when the data was collected.

> {{< alert title="Important" color="note" >}}

Reviewer: Should "Important" be changed to one of our standard style guide titles such as "Note"? Or added to the style guide?

Author: I think actually added to the style guide or maybe replace the Note in the style guide event. I'm never entirely sure what the implication for a note should be - does that tell the user "definitely read this" - or "read this for extra information". Let's discuss in team slack.

Author: nvm let's just change it to note

Reviewer: Haha okay either way!

> {{< alert title="Important" color="note" >}}

Author: ```suggestion
{{< alert title="Note" color="note" >}}
```

> Datasets are views, not materialized.
> If the underlying data matching the filter changes because data is deleted or more data is added, the dataset will also change.
> 
> Therefore the data you are viewing may change as you label and train on the dataset.

Reviewer: ```suggestion
The data you are viewing may change as you label and train on the dataset.
If the underlying data matching the filter changes because data is deleted or more data is added, the dataset will also change.
```
[opt] As a layperson, I didn't know what "materialized" means in this context. If that's a standard way to put it, for sure leave it in. But otherwise maybe this is clearer?

{{< /alert >}}

> You can label you data with

Reviewer: ```suggestion
You can label you data with:
```
[opt][dunno if this is better]

- [Bounding boxes](#bounding-boxes), which you can use to [train object detection models](/manage/ml/train-model/#train-a-model).

> - [Image tags](#image-tags), which you can use to [train object detection models](/manage/ml/train-model/#train-a-model).

Author: ```suggestion
- [Image tags](#image-tags), which you can use to [image classification models](/manage/ml/train-model/#train-a-model).
```

> - [Image tags](#image-tags), which you can use to [train object detection models](/manage/ml/train-model/#train-a-model).

Reviewer 2: Image tags are use to train image classification models

Author: good catch!

### Bounding boxes

You can create one or more bounding boxes for objects in each image.

> If you label an entire dataset, you can use these labels to [create object detection models](/manage/ml/train-model/#train-a-model).

Reviewer 2: ```suggestion
If you annotate an entire dataset, you can use these bounding boxes to [create object detection models](/manage/ml/train-model/#train-a-model).
```

> To add a bounding box, click on the image you want to add the bounding box.

Reviewer: ```suggestion
To add a bounding box, click on the image you want to add the bounding box to.
```

More information about the image will open up to the right side.
Select the **Bounding Box** option.

> Choose a label or create a new label to choose and then click on the image where you would like to add the bounding box and drag to where the bounding box should end.

Reviewer 2: Maybe "Choose an existing label or create.."

Reviewer: And maybe split into two sentences:
```suggestion
Choose an existing label or create a new label.
Click on the image where you would like to add the bounding box and drag to where the bounding box should end.
```

> {{<gif webm_src="../img/label-dog.webm" mp4_src="../img/label-dog.mp4" alt="Add bounding box label around dog in an image">}}

Reviewer 2: I think it should just be "Add a bounding box around the dog in an image"

> Repeat this with all images in your dataset.

Reviewer 2: I would add a note to say multiple bounding boxes can be added within an image

Author: I think I'll add that info to the first sentence in this section

> Repeat this with all images in your dataset.

Author: ```suggestion
Repeat this with all images in your dataset.
To see all the images that have a bounding box label, you can filter your dataset by selecting the label from the **Bounding box labels** dropdown in the **Filtering** menu.
```

> To delete a bounding box, click the cross next to the corresponding label.

Reviewer 2: Maybe a small note about being able to filter by bounding boxes similar to tags. 

You can filter the data in your dataset by selecting 'dog' from "Bounding box labels" dropdown which will show all images with dog bounding boxes.

Author: added above

### Image Tags

> You can use tags to [create classification models](/manage/ml/train-model/#train-a-model) for images.

Reviewer: ```suggestion
You can use tags to [create classification models](../../manage/ml/train-model/#train-a-model) for images.
```
[opt] Are we trying to do all relative links these days?

> To see all the images that have a bounding box label, you can filter your dataset by selecting the label from the **Bounding box labels** dropdown in the **Filtering** menu.

Reviewer 2: ```suggestion
To see all the images that have bounding boxes, you can filter your dataset by selecting the label from the **Bounding box labels** dropdown in the **Filtering** menu.
```

> For example, if you would like to create a model that identifies an image of a star in a set of images, tag each image with a star with a `star` tag.

Reviewer: ```suggestion
For example, if you would like to create a model that identifies an image of a star in a set of images, tag each image containing a star with a `star` tag.
```
Or similar, so it doesn't say "with a star" twice

The filter also needs to include images without the star tag or with another tag like `notstar`.

> If you add a `nostar` tag, you can filter the data in your dataset by adding the explicit tags `star` and `notstar` as a filter.

Reviewer 2: Maybe "If you add a `nostar` tag, you can filter the data in your dataset by selecting `star` and `notstar` from "Tags" dropdown in the filter"

Author: ```suggestion
If you add a `nostar` tag, you can filter the data in your dataset by selecting `star` and `notstar` from the **Tags** dropdown in the **Filtering** menu.
```

> If you add a `nostar` tag, you can filter the data in your dataset by adding the explicit tags `star` and `notstar` as a filter.

Reviewer: Stay consistent with either "nostar" or "notstar" (if I'm understanding correctly)

Alternatively you can use date ranges to filter for relevant data.

To tag an image, click on the image.
More information about the image will open up to the right side.
Select the **Image Tags** option.

> To add a bounding box, click on the image you want to add the bounding box to and, in the menu that opens, select the Bounding Box option.

Reviewer 2: We can delete the information about "To add a bounding box, click on the image you want to add the bounding box to and, in the menu that opens, select the Bounding Box option." since this is specific to just bounding boxes, while this section is about Image Tags

> To add a bounding box, click on the image you want to add the bounding box to and, in the menu that opens, select the Bounding Box option.

Author: ```suggestion
```

> Choose an existing tag or create a new tag to apply it to the image.

Author: ```suggestion
```

{{<gif webm_src="../img/label-dog.webm" mp4_src="../img/label-dog.mp4" alt="Add a bounding box around the dog in an image">}}

Repeat this with all images in your dataset.
To see all the images that have a bounding box label, you can filter your dataset by selecting the label from the **Bounding box labels** dropdown in the **Filtering** menu.

To delete a bounding box, click the cross next to the corresponding label.

### Image Tags

You can use tags to [create classification models](../../manage/ml/train-model/#train-a-model) for images.
For example, if you would like to create a model that identifies an image of a star in a set of images, tag each image containing a star with a `star` tag.
The filter also needs to include images without the star tag or with another tag like `notstar`.
If you add a `notstar` tag, you can filter the data in your dataset by selecting `star` and `notstar` from the **Tags** dropdown in the **Filtering** menu.
Alternatively you can use date ranges to filter for relevant data.

To tag an image, click on the image.
More information about the image will open up to the right side.
Select the **Image Tags** option.

> 

Author: ```suggestion
```
